pipewire/qubes-pw-module.c

```diff
@@ -835,6 +910,67 @@ static void vchan_ready(struct spa_source *source)
     }
 }
 
+static int vchan_reconnect_cb(struct spa_loop *loop,
+                                bool async,
+                                uint32_t seq,
+                                const void *data,
+                                size_t size,
+                                void *user_data)
+{
+    struct qubes_stream *stream = user_data;
+    stop_watching_vchan(stream);
+    return 0;
+}
+
+static void qdb_cb(struct spa_source *source)
+{
+    struct impl *impl = source->data;
+    int domid;
+
+    pw_log_debug("Received event from QubesDB");
+
+    domid = get_domid_from_qdb(impl->qdb);
+
+    if (domid != impl->domid) {
+        if (domid >= 0)
+            pw_log_info("Setting new peer domain ID %d", domid);
+        else if (domid == -ENOENT) // not an error, AudioVM unset
+            pw_log_info("AudioVM unset, disconnecting from %d", impl->domid);
+        else {
+            errno = -domid;
+            pw_log_error("Cannot obtain new peer domain ID: %m");
```

Comment:
Shouldn't it keep old `impl->domid` in this case?

Reply:
This is going to be either a negative value or an acceptable value. It will change in all the case on new audiovm so I don't think it matters.

Reply:
Ok, so you interpret failing to read as "no audiovm". I'm fine with this, but please adjust the message to be clear about it.